### PR TITLE
feat: add permaswap fetcher

### PR DESCRIPTION
**Pull request by ethever**

### CODE REVIEW

File: packages/oracle-node/manifests/data-services/main.json
Comment (cehali) [inline packages/oracle-node/manifests/data-services/main.json:1]: We don't use an automatic script for generating the main manifest as it provides too many changes. Please add new tokens manually.

Comment (cehali) [inline packages/oracle-node/manifests/data-services/main.json:1]: We don't use an automatic script for generating the main manifest as it provides too many changes. Please add new tokens manually.

**Comment by ethever** [comment]
thanks for reviewing. I have updated my pull request to address the requested changes.

**Comment by ethever** [comment]
any person help pls

**Review by cehali — CHANGES_REQUESTED** [review]
Details in comments

**Comment by ethever** [comment]
Thanks for reviewing. I have updated my pull request to address the requested changes.

**Comment by ethever** [comment]
Thanks for reviewing. I have updated my pull request to address the requested changes.

File: packages/oracle-node/test/fetchers/permaswap.spec.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const ardriveToArPriceResponse =
+  "../../src/fetchers/permaswap/example-ardrive-ar-response.json";
+
+const acnhToUsdcPriceResponse =
+  "../../src/fetchers/permaswap/example-acnh-usdc-response.json";
+
+jest.mock("axios");
+
+describe("permaswap fetcher", () => {
```
Comment (koonopek) [inline packages/oracle-node/test/fetchers/permaswap.spec.ts:17]: Please add also tests for failing paths like:
- missing value in db
- missing expected key in response from permaswap router
- permaswap router not answering

Comment (koonopek) [inline packages/oracle-node/test/fetchers/permaswap.spec.ts:17]: Please add also tests for failing paths like:
- missing value in db
- missing expected key in response from permaswap router
- permaswap router not answering

**Review by koonopek — DISMISSED** [review]

File: packages/oracle-node/src/fetchers/permaswap/example-pool-list-response.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (cehali) [inline packages/oracle-node/src/fetchers/permaswap/example-pool-list-response.json:1]: I think this file is not necessary in the code base

File: packages/oracle-node/test/fetchers/permaswap.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  });
+
+  afterAll(async () => {
+    await closeLocalLevelDB();
+  });
+
+  const sut = fetchers["permaswap"];
+
+  it("should properly fetch ARDRIVE price", async () => {
+    // Given
```
Comment (cehali) [inline packages/oracle-node/test/fetchers/permaswap.spec.ts:32]: You can remove those comments, I know they are in some files but they are not very informative

**Review by cehali — DISMISSED** [review]
Minor changes in comments and please add additional tests

File: packages/oracle-node/src/fetchers/permaswap/PermaswapFetcher.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      throw new Error(`${requestId} not found in ${this.name} fetcher`);
+    }
+    return pool;
+  }
+
+  protected getPairedTokenPrice(assetId: string): number {
+    let { pairedToken } = this.getPool(assetId);
+
+    const lastPriceFromCache = getLastPrice(pairedToken);
+    console.error({ assetId, lastPriceFromCache });
```
Comment (hatskier) [inline packages/oracle-node/src/fetchers/permaswap/PermaswapFetcher.ts:65]: Can you please remove this console.error? Or use the this.logger.error instead?

Comment (hatskier) [reply packages/oracle-node/src/fetchers/permaswap/PermaswapFetcher.ts:65]: I think it's better to remove :)

**Review by hatskier — APPROVED** [review]
Approved with a minor comment, will merge after fixing

**Comment by ethever** [comment]
Thanks for reviewing. I have updated my pull request to address the requested changes.

**Comment by ethever** [comment]
any progress?

**Review by hatskier — APPROVED** [review]

**Pull request merged**